app/src/main/java/com/tsu/wordsfactory/app/WordsApplication.kt

```diff
@@ -0,0 +1,59 @@
+package com.tsu.wordsfactory.app
+
+import android.app.Application
+import android.content.Context
+import com.google.gson.GsonBuilder
+import com.tsu.midtermexam3.data.Database
+import com.tsu.midtermexam3.data.NotesRepository
+import com.tsu.wordsfactory.repository.WordsDatasourceImpl
+import com.tsu.wordsfactory.repository.WordsRepository
+import com.tsu.wordsfactory.repository.WordsRepositoryImpl
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.create
+
+class WordsApplication: Application() {
+    private companion object {
+        var BASE_URL = "https://api.dictionaryapi.dev/api/v2/entries/en/"
+    }
+
+    lateinit var wordsRepository: WordsRepository
+
+    lateinit var retrofit: Retrofit
```

Comment:
This variables shouldn't be inside application. It's better to create `object Network` and create retrofit there